### PR TITLE
Clean up yaml and add a yaml-lint check

### PR DESCRIPTION
**Pull request by wlach**
* Add yamllint check and fix various minor issues it found
* Remove unneeded quotes from repositories.yaml

### CODE REVIEW

**Review by fbertsch — CHANGES_REQUESTED** [review]
@wlach did you mean to add this to CI? It would be very useful to ensure that the files have been linted on every commit.

**Comment by wlach** [comment]
> @wlach did you mean to add this to CI? It would be very useful to ensure that the files have been linted on every commit.

Oops yes! This was supposed to be part of the pull request but it somehow got lost. Added it back.

File: Makefile [hunk 1]
```diff
@@ -24,6 +24,7 @@ clean-pyc:
 
 lint: build
 	docker-compose run app flake8 --max-line-length 100 .
+	docker-compose run app yamllint repositories.yaml .circleci
```
Comment (fbertsch) [inline Makefile:27]: Do we need to add this as a dependency?

Comment (fbertsch) [reply Makefile:27]: I didn't know that. Probably not a big deal either way then

**Review by fbertsch — APPROVED** [review]
LGTM, thank you! We should do this to bigquery-etl tests and metadata as well

File: Makefile (hunk 1 above)
Comment (wlach) [reply Makefile:27]: It's easy to add, so I added it. Will save us if glean ever loses the dep.

**Pull request merged**